tests/components/climate/test_generic_thermostat.py

```diff
@@ -52,24 +59,40 @@ def tearDown(self):  # pylint: disable=invalid-name
     def test_setup_missing_conf(self):
         """Test set up heat_control with missing config values."""
         config = {
-            'name': 'test',
-            'target_sensor': ENT_SENSOR
+            'name': 'test'
         }
         with assert_setup_component(0):
             setup_component(self.hass, 'climate', {
                 'climate': config})
 
-    def test_valid_conf(self):
+    def test_setup_valid_conf(self):
+        """Test set up generic_thermostat with valid config values."""
+        self.assertTrue(
+            setup_component(self.hass, 'climate',
+                            {'climate': {
+                                'platform': 'generic_thermostat',
+                                'name': 'test',
+                                'heater_control': ENT_SWITCH_HEAT,
+                                'target_sensor': ENT_SENSOR
+                                }})
+        )
+
+    @mock.patch('logging.Logger.error')
```

Comment:
What are we testing here? Just testing that something is logged? 🤔  we shouldn't do any logging on setup.

If you want to test logging, use `caplog` fixture

Reply:
I just reused a previous test that was doing the same and adapted it to the current needs. The test was trying to make sure that with invalid settings you get an error message printed.

Reply:
If a config value is invalid, it should be flagged by validation done by `PLATFORM_SCHEMA`